scripts/execwhacker.py

```diff
@@ -56,11 +67,75 @@ class ProcessSource(Enum):
 
 processes_checked = Counter(f"{cryptnono_metrics_prefix}_execwhacker_processes_checked_total", "Total number of processes checked", ["source"])
 processes_killed = Counter(f"{cryptnono_metrics_prefix}_execwhacker_processes_killed_total", "Total number of processes killed", ["source"])
+processes_missed = Counter(f"{cryptnono_metrics_prefix}_execwhacker_processes_missed_total", "Total number of processes that independently exited whilst being processed", ["source"])
+processes_allowed = Counter(f"{cryptnono_metrics_prefix}_execwhacker_processes_allowed_total", "Total number of processes allowed", ["source", "allowedby"])
+
+unexpected_errors = Counter(f"{cryptnono_metrics_prefix}_execwhacker_unexpected_errors_total", "Total number of unexpected errors, usually indicates a programming or configuration error")
 
 
-def kill_if_needed(banned_strings_automaton, allowed_patterns, cmdline, pid, source):
+def log_and_kill(pid, cmdline, b, source, lookup_container):
+    """
+    Attempt to lookup the container details for a given PID, then log and kill it
+
+    This makes an external blocking call to lookup the container details
+
+    Returns True if the process was killed, False if it was not found
+    """
+
+    cid = None
+    log = logging.bind(pid=pid, cmdline=cmdline, matched=b, source=source.value)
+
+    if lookup_container:
```

Comment:
What do you think about adding a prometheus metric here that's collecting a histogram of how *long* the lookup actually takes? I think it would be very helpful for us to see, mostly so we know if it's actually leading to *more* races than before.

Reply:
If this gets fast enough, I'll be tempted for us to remove the multi threading so the code can be simpler :D